query/src/main/scala/filodb/query/exec/InProcessPlanDispatcher.scala

```diff
@@ -0,0 +1,60 @@
+package filodb.query.exec
+
+import scala.concurrent.duration.FiniteDuration
+
+import com.typesafe.config.ConfigFactory
+import monix.eval.Task
+import monix.execution.Scheduler
+import monix.reactive.Observable
+
+import filodb.core.DatasetRef
+import filodb.core.Types.ColumnId
+import filodb.core.metadata.Dataset
+import filodb.core.store.{ChunkScanMethod, ChunkSource, ChunkSourceStats, PartitionScanMethod, RawPartData,
+  ReadablePartition, ScanSplit}
+import filodb.query.{QueryConfig, QueryResponse}
+
+/**
+  * Dispatcher which will make a No-Op style call to ExecPlan#excecute().
+  * Goal is that Non-Leaf plans can be executed locally in JVM and make network
+  * calls only for children.
+  * @param dataset to be used by ExecPlan#execute
+  */
+case class InProcessPlanDispatcher(dataset: Dataset) extends PlanDispatcher {
+
+  override def dispatch(plan: ExecPlan)(implicit sched: Scheduler,
+                                        timeout: FiniteDuration): Task[QueryResponse] = {
+
+    // Empty query config, since its does not apply in case of non-leaf plans
+    val queryConfig: QueryConfig = new QueryConfig(ConfigFactory.empty())
```

Comment:
We should get query-config from a static context. Should not be re-creating each time.